eva/optimizer/plan_generator.py

```diff
@@ -48,37 +49,29 @@ def build_optimal_physical_plan(self,
     def optimize(self, logical_plan: Operator):
         optimizer_context = OptimizerContext()
         memo = optimizer_context.memo
-        grp_expr = optimizer_context.xform_opr_to_group_expr(
-            opr=logical_plan,
-            copy_opr=False
+        grp_expr = optimizer_context.add_opr_to_group(
+            opr=logical_plan
         )
         root_grp_id = grp_expr.group_id
+        root_expr = memo.groups[root_grp_id].logical_exprs[0]
 
-        """
-        for g in optimizer_context.memo._groups:
-            for expr in g._logical_exprs:
-                print(expr)
-        """
         # TopDown Rewrite
         optimizer_context.task_stack.push(
-            TopDownRewrite(grp_expr, optimizer_context))
+            TopDownRewrite(root_expr, RulesManager().rewrite_rules,
+                           optimizer_context))
         self.execute_task_stack(optimizer_context.task_stack)
-        # Update the group expression
-        grp_expr = memo.groups[root_grp_id].logical_exprs[0]
+
         # BottomUp Rewrite
+        root_expr = memo.groups[root_grp_id].logical_exprs[0]
         optimizer_context.task_stack.push(
-            BottomUpRewrite(grp_expr, optimizer_context))
+            BottomUpRewrite(root_expr, RulesManager().logical_rules,
```

Comment:
Why top down rewrite_rules and bottom up logical_rules?

Reply:
Will fix this.